### ISSUE TITLE
Select does not pass all jest-axe accessibility checks

### ISSUE THREAD

**Issue by halocline** [grommet/grommet#5501]
### Expected Behavior

Select should not have accessibility violations and pass this test:
```
test('should not have accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <Select options={['one', 'two', 'three']} a11yTitle="test" />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();

    expect(container.firstChild).toMatchSnapshot();
  });
```

### Actual Behavior

Running latest `jest-axe` (`^5.0.0`) which relies on `axe-core` (`^4.2.0` rule set) catches the following accessibility error:
![Screen Shot 2021-08-02 at 1 49 46 PM](https://user-images.githubusercontent.com/1756948/127915654-341f34d8-ab1b-422b-9cf1-2f19a1eb06be.png)

**Background**

- Current Select places an `<input />` nested within a `<button />`
![Screen Shot 2021-08-02 at 1 54 06 PM](https://user-images.githubusercontent.com/1756948/127916152-04f581a2-ffb0-4658-9d4d-129c054532fb.png)
- According to [Dequeue & WCAG 4.1.2](https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=axeAPI) "interactive controls must not have focusable descendants." 
- Why it matters: "Focusable elements with an interactive control ancestor (any element that accepts user input such as button or anchor elements) are not announced by screen readers and create an empty tab stop. That is, you could tab to the element but the screen reader will not announce its name, role, or state."

### Steps to Reproduce

- Modify `package.json` to use `"jest-axe": "^5.0.1"`
- `yarn install`
- `yarn test`

### Your Environment

- Grommet version: 2.17.4
- Browser Name and version:
- Operating System and version (desktop or mobile):
